### DISCUSSION TITLE
blog/2023-02-15-causal-inference-with-bayesian-models/

### DISCUSSION THREAD

**Discussion by giscus** [ASKurz/blogdown5#9] [Comments]
# blog/2023-02-15-causal-inference-with-bayesian-models/

Part 4 of the GLM and causal inference series.

https://timely-flan-2986f4.netlify.app/blog/2023-02-15-causal-inference-with-bayesian-models/

**Comment by sjwild** [comment]
Overall a good post, though definitely more applied than previous posts.
* I find this post a bit harder to follow compared to the previous, more theory-laden posts. The reasons is that the code chunks are closer together, and some of the explanatory text is easy to overlook because it goes code chunk, output, line of text, code chunk, output, line of text. The lines of text get buried because the code chunks and outputs are bigger. Not sure if there is an easy fix, but it is something I noticed.
* You could consider include a small tibble or paragraph at the beginning of the post with the treatment effect estimates for the frequentist models from the earlier posts. It makes it easier for readers rather than having to go back to find them. 
* the exponential distribution uses a _rate_ or a _scale_ parameter depending on the parameterization. Might be worth mentioning in a footnote, because it can get confusing across software.
* When you are describing why you made use of the exponential distribution, I think it is worth explaining that the exponential distribution cannot be negative. It's implied and obvious for us, maybe less so for the less well-initiated.
* In the initial ANCOVA model, it is probably worth mentioning why you expect weight to be so highly correlated. This wasn't an issue for the frequentist models (because no prior), but here I think the extra sentence or two will be helpful context for those who haven't much experience with Bayesian inference and priors.
* Not sure how much detail you want to get into, but it might be worth directing interested readers to some resources about MCMC (e.g., McElreath lectures). It depends on how knowledgeable you think your audience is, and whether you think newbie/potential Bayesian will read this. It looks like this post is directed towards more knowledgeable Bayesians, so the resources and/or explanation might not be necessary.
* Consider explaining why you are going with `brm(... post ~ 0 + Intercept + ..., ...)` compared to how the models were entered in the previous posts. Or just handwave it away for now. 
* Disappointed there is no `plogis()`, and instead you use `inv_logit_scaled()` :-p
* You mention "You’ll note many of the parameters from our Bayesian fit4 are more conservative compared to its frequentist counterpart.", But unless I go back to post 3, I'm not sure how different they are. I think it's worth a plot here featuring estimates from both posts.
* Ditto for the last plot, where you showcase the  diversity among the $\tau_i$'s.

**Comment by isabellaghement** [comment]
Hey Solomon, 

I jumped right off to post # 4 in your series.  Stephen beat me to it so I will try to provide some feedback that doesn’t overlap with his. 😁

Before providing any feedback, I just wanted to reiterate how much I enjoy your easy flowing writing and also the opportunity to learn from the feedback provided by our group.  We all come at this from different perspectives, which hopefully covers a lot of ground on behalf of your intended audience.

1) At the top of your post, can you include a brief paragraph to explain how the Bayesian modelling paradigm differs from the Frequentist paradigm?  Something like:

*“In the Frequentist modelling paradigm, we estimate the model unknowns (i.e., the model parameters) solely on the basis of the data. In the Bayesian modelling paradigm, we estimate these unknowns by combining two sources of information: 1) prior information about their plausible values derived based on previous literature findings, our subject-matter driven expectations, etc. and 2) the data.  Not only does the Bayesian modelling paradigm offer more flexibility in estimating these unknowns when the prior information supplements what transpires from the data alone, it also affords a more natural interpretation of model-based inferences. When we lack any prior information, the Bayesian modelling paradigm will mimic the Frequentist one in terms of findings, though still providing a more natural interpretation.”

2)  Your explanation of how you justified your choice of priors deserves some motivation:  *“The Bayesian modelling paradigm requires careful consideration and justification of what prior information is included in the model.  A good Bayesian modelling workflow should include a prior sensitivity analysis, which would investigate the sensitivity of the model findings to the specification of the prior information.”* 

3) Can you explain to your readers when and why they should consider switching from the Frequentist to the Bayesian modelling paradigm? Should they always use the Bayesian paradigm? 

4) Given how long the post is and how much ground it covers, it seems a good idea to me to give the readers an upfront but more specific warning of what they can expect to find in the post. Something like this added at the top of the post should do: 

*”In this post, we will revisit the Gaussian and Binomial ANOVA-type models in the previous posts, this time estimating the ATE using a Bayesian rather than a Frequentist modelling paradigm.  We will learn how to specify these models using Bayesian notation, how to fit these models using the brms package, how to compute and interpret the ATE, etc.”* 

This way, the readers will have a mental map of what’s to come rather than wade through the tedious details to infer that map. 

5) I agree with Stephen that having a side-by-side table showing the Frequentist and Bayesian Gaussian and Binomial models, Frequentist and Bayesian ATEs, etc., will be helpful.  Contrasts between what is familiar to the reader and what is unfamiliar are great tools when it comes to learning something new by building on something old. 

6) The issue of a more natural interpretation for the ATEs in models fitted within a Bayesian modelling paradigm needs to be properly highlighted.  Can you explain to your readers that they can say things like “there is a 95% probability that the true value of the ATE falls in the range ___ to ___” (which is a much stronger and more naturally interpretable statement than the Frequentist counterpart: “we are 95% confident that the true value of the ATE falls in the range ___ to ___)?  

Getting the numbers is the easy part; knowing how to interpret the numbers is typically overlooked in blog posts. As an applied statistician, I tend to be more interested than most in the interpretation of the numbers. 

 7) Can you define the ATE for each of your two models explicitly so that readers know what you are trying to estimate? Maybe even make the point that what you are trying to estimate is the same in both the Frequentist and the Bayesian framework? 

8) In your computation of ATE (regardless of model fitting framework), are you averaging only over the subjects represented in the sample (“narrow inference”)? If yes, can you comment for the readers on whether it would be possible to average over all subjects in the population (“broad inference”) and how? 

9) When you introduce the Binomial ANOVA-type model with n = 1 (i.e., one trial per subject), can you clarify for your readers that this is the same as a Bernoulli ANOVA-type model? Doesn’t brms have a bernoulli family that could be used to fit the model? If yes, the readers should know that is an option available to them. 

10) Speaking of Bernoulli, is the ATE in your post defined as the average of differences of the type E(Y_i = 1|Treatment = 1) - E(Y_i  = 1|Treatment = 0) where i = 1, …, n and n is the number of subjects?  If yes, the two expected values involved in the i-th difference are probabilities that Y_i = 1  when Treatment = 1 or 0, respectively, so they can be denoted as p_i(11) and p_i(10), say.  I confess that I don’t get it why there is another expected value applied to them: E(p_i(11)) and E(p_i(00))? Where does this expected value come from? Should it even be there? Unless you are now integrating somehow over all subjects in the population? But I didn’t see you take a large random sample of these subjects from the underlying population when computing the ATE?  Can you clarify in your post what is going on here? 

11) Because there was no explicit definition provided for the ATE that you would compute for the Gaussian/Binomial model, I really struggled to understand why you would **group by draw** when computing your ATE values.  

For example, in the Binomial model, if you are computing the difference between E(Y_i = 1| Treatment = 1) and E(Y_i = 1| Treatment = 0) for each subject first and then averaging these differences across subjects, I would have expected the grouping to be done by subject and then the averaging to be done across draws within subject separately for each treatment to get each of these two expected values for a given subject, etc.  Can you clarify this for your readers. 

12) The R syntax for computing the ATEs where you pivot and join, etc, is really complicated. Can you preface the first time you introduce that syntax what the steps involved are so the reader understands the complicated code? 

That’s all for now - if anything is not clear, please just let me know. 

Cheers, 

Isabella

**Comment by ASKurz** [comment]
Okay @sjwild and @isabellaghement, I’ve updated the post based on your feedback. Y’all both gave detailed feedback, so I’ll make some broader comments, rather than use a list format.

Thank you both for some of your kind words. I really appreciate the encouragement.

I added in a little more explanatory prose throughout. One example was in response to Isabella’s suggestions for a more informative intro. Another examples was just increasing the word count between some of the code blocks, as per Steve’s concerns the sections blurred together at times.

I also added in more technical explanations, such as briefly introducing the special `0 + Intercept` syntax, or explaining why you would (or perhaps sometimes wouldn’t) group by draw before summarizing. 

You both made several comments about making it easier to contrast the frequentist and Bayesian results, or comments requesting more explanations for beginner Bayesians. After sitting with those suggestions for a while, I’ve decided not to add those kinds of features. This blog post, and this overall series, is not intended for beginner frequentist or Bayesians. It’s probably not advanced, but it’s definitely at least intermediate. I actually think beginner Bayesians should be turned off from this post; their attention should really be centered on more foundational materials. As to the frequentist/Bayesian comparisons, I can totally understand why readers might want more of that, but those kinds of comparisons really aren’t the goal of this post. The goal, here, is to teach people how to use the standardization approach, not to compare/contrast frequentist and Bayesian approaches for standardization. 

Again, I really appreciate your feedback. It helped me clean up the post.
